server/src/main/java/org/opensearch/cluster/decommission/DecommissionStatus.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.decommission;
+
+/**
+ * An enumeration of the states during decommissioning
+ */
+public enum DecommissionStatus {
+    /**
+     * Decommission process is initiated, and to-be-decommissioned leader is excluded from voting config
+     */
+    INIT("init", 0),
+    /**
+     * Decommission process has started, decommissioned nodes should be removed
+     */
+    IN_PROGRESS("in_progress", 1),
+    /**
+     * Decommission action completed
+     */
+    SUCCESSFUL("successful", 2),
+    /**
+     * Decommission request failed
+     */
+    FAILED("failed", -1);
+
+    private final String status;
+    private final int stage;
```

Comment:
-1 will always be assigned to `FAILED` state. For any other stages we may introduce in the flow in future will and should always be in suitable order (eg. 1->2->3->4->...) depending on where that stage will be. Example -> If we introduce a draining stage, it can sit after INIT stage and in that case, stage of progress will be 2 and successful will be 3 and draining could be 1. 
This won't be breaking as this change is only used to assert the order it is coming in. (Also as long as FAILED is -1, we should be good)

Reply:
> rename stage to code.

With above logic related to stage/code, `stage` is looking is more appropriate here due to its incremental nature. Let me know if you think otherwise

Reply:
You are storing the status in cluster state metadata. You have added fromStage method, not sure how you are planning to use it. The same example you took, lets say we add a new stage DRAINING in between, now 2 will point to DRAINING instead of SUCCESSFUL and fromStage conversion would depend on which node it is executing as it would have different behavior depending on which node has latest code.

Reply:
> You have added fromStage method, not sure how you are planning to use it.

Reply:
> The same example you took, lets say we add a new stage DRAINING in between, now 2 will point to DRAINING instead of SUCCESSFUL and fromStage conversion would depend on which node it is executing as it would have different behavior depending on which node has latest code.

Removing this toStage method seems to resolve this issue, right? Also, when can it happen that same nodes in cluster have different code?